### PR TITLE
Handle unseen aspects from gossip list not just authoring list

### PR DESCRIPTION
**Pull request by zippy**
## PR summary

There's a bug where sim2h isn't keeping track of unseen aspects in it's model of which aspects exist in the space from gossiping data only from authoring data.  This means that in this situation:  1) all the nodes that have some aspect data go off line, 2) a new node comes on-line,  3) an old node which had data but was not the author, then sim2h would not know to send the data to the new node.

This PR adds a test and refactors the handle_unseen code for use in both situations.

## testing/benchmarking notes

It also identified that tryorama was using the in-memory storage for stress tests, which would cause this case to fail because the old non-authoring node coming back on line would have lost it's holding list.

The new `gossip.ts` in the stress-tests creates a regression test.

## followups

( any new tickets/concerns that were discovered or created during this work but aren't in scope for review here )

## changelog

- [ ] if this is a code change that effects some consumer (e.g. zome developers) of holochain core,  then it has been added to [our between-release changelog](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md) with the format 

```markdown
- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)
```

## documentation

- [ ] this code has been documented according to our [docs checklist](https://hackmd.io/@freesig/Hk9AmKJNS)

### CODE REVIEW

**Comment by zippy** [comment]
stress test won't pass until tryorama updates get released

File: crates/sim2h/src/lib.rs [hunk 1]
```diff
@@ -558,34 +586,15 @@ impl Sim2h {
...
-                }
+                self.handle_unseen_aspects(uri, space_address, agent_id, &list_data);
                 Ok(())
             }
             WireMessage::Lib3hToClientResponse(Lib3hToClientResponse::HandleGetGossipingEntryListResult(list_data)) => {
                 debug!("GOT GOSSIPING LIST from {}", agent_id);
                 if (list_data.provider_agent_id != *agent_id) || (list_data.space_address != *space_address) {
                     return Err(SPACE_MISMATCH_ERR_STR.into());
                 }
+                self.handle_unseen_aspects(uri, space_address, agent_id, &list_data);
```
Comment (struktured) [inline crates/sim2h/src/lib.rs]: This is the line that actually fixes it, correct?

File: stress-test/src/config.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ const network =
   : networkType === 'sim2h'
   ? {
     type: 'sim2h',
-    sim2h_url: 'ws://localhost:9002'
+    sim2h_url: 'wss://localhost:9002'
```
Comment (struktured) [inline stress-test/src/config.ts]: :+1:

File: crates/sim2h/src/lib.rs (hunk 1 above)
Comment (zippy) [reply crates/sim2h/src/lib.rs]: Yep

**Pull request merged**